MOCC/src/configurations/Comm-Layer/src/tests/test_server.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+import sys
+import uuid
+sys.path.insert(0, '..')
+
+from server import ServerBase, Command
+class Server(ServerBase):
+    """
+    Sample server.
+    """
+    @Command
+    def cmd1(self, arg1, arg2):
+        ret_val = {
+            'arg1': arg1,
+            'arg2': arg2,
+        }
+        return ret_val
+
+
+@pytest.fixture
+def client():
+    server = Server(__name__)
+    server.config['TESTING'] = True
+    client = server.test_client()
+
+    yield client
+
+
+def test_cmd1(client):
+    data = {
+        'arg1': str(uuid.uuid4()),
+        'arg2': str(uuid.uuid4()),
+    }
+    rv = client.post('/cmd1')
```

Comment:
hey Arpit, sorry it took me so long to answer.
Here you forgot to pass `data` to the POST request.
the call should be: `rv = client.post('/cmd1', json=data)`
with that you receive the data in [here](https://github.com/mars-planet/mars_city/pull/62/commits/d30bc41b5fa8ece37ace76bebc41f45efc8787ec#diff-4b8a31c3ed7b9339b5f506009e73269cR533)

Reply:
This is the error that I get when I put `json=data`
>E       TypeError: __init__() got an unexpected keyword argument 'json'

But when I change it to `data=data` this is what I get
![screen shot 2018-07-21 at 12 18 28 am](https://user-images.githubusercontent.com/9051659/43019742-9f4e1d30-8c7b-11e8-923d-1c24d5ab2416.png)

Reply:
Did you check the version of your libraries? Are the stable versions?
What about Python's version?

Reply:
You shouldn't use the `data` parameter.
If you do, then you need to change the code where you read from `requests.json`

Reply:
According to this https://stackoverflow.com/a/46991833/4014694 it's fine to do so

Reply:
> If you do, then you need to change the code where you read from requests.json
What change needs to be made? From what I can see the request object has pretty much only the URL

Reply:
The thread you linked to refers to the `requests` library, but you're sending the request using the test client from Flask.
I don't know exactly what changes you need to make, if I did, I'd just do it myself.
If you feel you cannot continue with this, I can pick up the task from here.

Reply:
That would be beneficial..